app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Proxy files through application with authentication.
+# This avoids making the S3 bucket where files are stored public
+
+# This controller is heavily customized for Exports. When Rails 7 is released, these changes can get incorporated into the Downloads controller itself.
+# Rails 7 is expected to have features related to making it possible to stream files without having to overwrite this controller.
+# This will allow us to remove the custom authentication and delete job queuing.
+
+# Overwriting
+# https://github.com/rails/rails/blob/6-1-stable/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
+class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
```

Comment:
For a missing download (for example if the download has expired and has been deleted) I just want to verify that this will show a 404 page in production.

![Screen Shot 2021-05-01 at 7 49 23 AM](https://user-images.githubusercontent.com/24628687/116784499-c179f400-aa51-11eb-88f6-797ddf4db708.png)

Reply:
Yep I just confirmed that it does. You can test by setting in config/environments/development.rb
`config.consider_all_requests_local = true`

(thank you @Bialogs for that)